src/FixedDataTable.js

```diff
@@ -1338,13 +1340,16 @@ var FixedDataTable = createReactClass({
     if (!this._isScrolling) {
       this._didScrollStart();
     }
+
+    var newState = {};
+
     var onHorizontalScroll = this.props.onHorizontalScroll;
     if (onHorizontalScroll ? onHorizontalScroll(scrollPos) : true) {
-      this.setState({
-        scrollX: scrollPos,
-      });
+      newState.scrollX = scrollPos;
+      this.setState(newState, this._didScrollStop);
+    } else {
+      this._didScrollStop();
```

Comment:
@wcjordan , we don't update `scrollX` if `onHorizontalScroll` returns false. This means we can directly call `didScrollStop`